core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -73,147 +74,131 @@ public JobUpdateTask(
     this.currentJob = currentJob;
     this.jobManager = jobManager;
     this.jobUpdateTimeoutSeconds = jobUpdateTimeoutSeconds;
+    this.runnerType = jobManager.getRunnerType().toString();
```

Comment:
We had confusion and almost a bug before around `Job#runner` being a string, which I _think_ #575 has fixed, but I feel this is still very error-prone. I think I want to also refactor `Job#runner` to be the enum type instead of stringly-typed programming, I'll see how much that change hurts but it should probably be another PR anyway.

Reply:
The enum matter is raised in #651.